### ISSUE TITLE
Pod Preset does not inject the init container

### ISSUE THREAD

**Issue by wstrange** [kubernetes/kubernetes#55410]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

/sig cluster-lifecyle

**What happened**:

I created a Pod Preset to inject an environment variable. The Pod container gets properly injected, but the init container does not. 

A small sample is here:

https://gist.github.com/wstrange/b9c1adc2139dab8f9b554a9d21b30bdd

**What you expected to happen**:

The init container would get the pod preset injected 

**How to reproduce it (as minimally and precisely as possible)**:

Run kubectl create -f on the following gist:
https://gist.github.com/wstrange/b9c1adc2139dab8f9b554a9d21b30bdd

Reproduced here:

```
apiVersion: settings.k8s.io/v1alpha1
kind: PodPreset
metadata:
  name: test
spec:
  selector:
    matchLabels:
      app: test
  env:
    - name: SETME
      value: "This should be set in the init container"

---
apiVersion: v1
kind: Pod
metadata:
  name: test
  labels:
    app: test
spec:
  initContainers:
  - name: init
    image: busybox
    args: ["env"]
  containers:
    - name: test
      image: busybox
      args: [ "sleep", "1000"]

```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):  
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.3", GitCommit:"f0efb3cb883751c5ffdbe6d515f3cb4fbe7b7acd", GitTreeState:"clean", BuildDate:"2017-11-09T07:26:38Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.0", GitCommit:"0b9efaeb34a2fc51ff8e4d34ad9bc6375459c4a4", GitTreeState:"dirty", BuildDate:"2017-10-17T15:09:55Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration: minikube 0.23.0
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:
Started with:
```

minikube start \
--extra-config=apiserver.Admission.PluginNames="Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,GenericAdmissionWebhook,ResourceQuota,PodPreset"
```

**Comment by wstrange** [comment]
My use case: I have an a git init container that checkout out configuration. This needs to run before the main container (hence why it is an init container).
I'd like to be able to simply annotate the pod with a label `configsource: git` and have various environment variables (GIT_REPO, etc) supplied to the init container

**Comment by wstrange** [comment]
Update from Slack discussion:

Supporting Init containers is in the Beta proposal for PodPreset. PodPreset resource is going to be moved from core to service-catalog. We have an implementation based on Initializers, but looks like Initializers are not going to Beta soon and we might have to reimplement them using Webhooks.

**Comment by DanielJonesEB** [comment]
It'd be nice if this worked, I suspect it would help the folks over at Knative Build implement support for passing HTTP_PROXY env vars and CA cert secrets to build steps.

**Comment by ncdc** [comment]
We have a customer interested in seeing this fixed - if we submit a PR, are there objections to merging it?

**Comment by ncdc** [comment]
cc @pmorie

**Comment by dankelleher** [comment]
This is also affecting me - I have an init container that loads data from AWS Secrets and stores them in a volume that is accessed by the main container. The init container needs credentials that are passed via a podpreset

**Comment by marouanabbessi** [comment]
any ETA when support for initcontainers will be provided by PodPreset ?

**Comment by ncdc** [comment]
This was fixed in 1.14.
